mbtiles/scripts/cli.py

```diff
@@ -285,6 +309,31 @@ def mbtiles(
                 src.crs, "EPSG:4326", src.bounds[::2], src.bounds[1::2]
             )
 
+            # cutlines must be transformed from CRS84 to src pixel/line
+            # coordinates and then formatted as WKT.
+            if cutline is not None:
+                geoms = [shape(f["geometry"]) for f in cutline]
+                union = unary_union(geoms)
+                if union.geom_type not in ("MultiPolygon", "Polygon"):
+                    raise click.ClickException("Unexpected cutline geometry type")
+                west, south, east, north = union.bounds
+                cutline_src = shape(
+                    transform_geom("OGC:CRS84", src.crs, mapping(union))
+                )
+                invtransform = ~src.transform
+                shapely_matrix = (
+                    invtransform.a,
+                    invtransform.b,
+                    invtransform.d,
+                    invtransform.e,
+                    invtransform.xoff,
+                    invtransform.yoff,
+                )
+                cutline_rev = shapely.affinity.affine_transform(
+                    cutline_src, shapely_matrix
+                )
+                warp_options["cutline"] = shapely.wkt.dumps(cutline_rev)
```

Comment:
This is the second aspect of using a cutline: we use it as a warp cutline to fence pixels in/out.